dataproc/create_hail_cluster.ipynb

```diff
@@ -0,0 +1,865 @@
+   "source": [
+    "TEMP_BUCKET_CMD_OUTPUT = !terra resolve --name=ws_files_autodelete_after_two_weeks\n",
+    "TEMP_BUCKET = TEMP_BUCKET_CMD_OUTPUT[0]\n",
+    "TEMP_BUCKET"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c05391eb-f42c-4b5f-995b-8452db1f292d",
+   "metadata": {},
+   "source": [
+    "## Define an autoscaling policy\n",
+    "\n",
+    "Configure Dataproc [autoscaling](https://cloud.google.com/dataproc/docs/concepts/configuring-clusters/autoscaling) to automatically and dynamically scale the number of worker VMs in Dataproc clusters to meet workload demands.\n",
+    "\n",
+    "People will likely have many different autoscaling policies, since some jobs will run best with different numbers of primary workers that will not be preempted."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "35d8725e-b32e-4cbe-8285-ab7defe7c13d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%writefile two_worker_autoscaling_policy.yaml\n",
+    "\n",
+    "workerConfig:\n",
+    "  # Best practice: keep min and max values identical for primary workers\n",
+    "  # https://cloud.google.com/dataproc/docs/concepts/configuring-clusters/autoscaling#avoid_scaling_primary_workers\n",
+    "  minInstances: 2\n",
+    "  maxInstances: 2\n",
+    "secondaryWorkerConfig:\n",
+    "  maxInstances: 50\n",
+    "basicAlgorithm:\n",
+    "  cooldownPeriod: 4m\n",
+    "  yarnConfig:\n",
+    "    scaleUpFactor: 0.05\n",
+    "    scaleDownFactor: 1.0\n",
+    "    gracefulDecommissionTimeout: 1h"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "49cd9df5-5988-4fd2-b3a6-9c0d40587ea8",
+   "metadata": {},
+   "source": [
+    "Import the autoscaling policy, if it does not yet exist."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ce433f11-b715-42f0-8ac5-17d87901b6c0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!gcloud dataproc autoscaling-policies describe two_worker_autoscaling_policy --region=us-central1 || \\\n",
+    "    gcloud dataproc autoscaling-policies import two_worker_autoscaling_policy \\\n",
+    "        --source=two_worker_autoscaling_policy.yaml \\\n",
+    "        --region=us-central1"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4138e077-0d37-4ae2-8bd5-6a44d3dc70ee",
+   "metadata": {},
+   "source": [
+    "## Define the Hail initialization script\n",
+    "\n",
+    "This script is based on the logic in [hailctl](https://hail.is/docs/0.2/cloud/google_cloud.html#hailctl-dataproc) version 0.2.108."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a83c0ee2-39fc-47e8-babb-46949b5c05ee",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%writefile init_tvc_hail_cluster.py\n",
+    "#!/opt/conda/default/bin/python3\n",
+    "\n",
+    "# This script is based on gs://hail-common/hailctl/dataproc/0.2.108/init_notebook.py with all JupyterLab configuration removed.\n",
```

Comment:
nit: line is a bit too long, consider adding newline so the comment displays well on GitHub